### PR TITLE
Adds an extra effect to inaprovaline

### PR DESCRIPTION
**Pull request by aceedex**
This PR adds a new effect to inaprovaline. When someone with inaprovaline in their system enters critical and they are buckled they will now heal all main damage types, 2 damage per tick. Its also a little strange but no healing is done if the person is also strapped to a any vehicle including the medical malpractice because god help me if I ever make that thing good again.
Im hoping this change will help when it comes to keeping people alive when in transport towards medbay who have conditions that deal damage over time like liver damage or severe blood loss.

:cl:
 * rscadd: inaprovaline now moderately heals damage so long as the user is in critical and buckled into something

### CODE REVIEW

**Comment by Kurfursten** [comment]
Inaprovaline can be found in something like 30% of O2 lockers, so I don't like it being a powerful combat stimulant drug that can reliably pull you back out of crit if you go down due to lasers

**Comment by aceedex** [comment]
Doesnt being put into crit make you fall off the janicarts and secways? 
well, its not difficult to add those to the exceptions too

**Comment by Kurfursten** [comment]
if you're already buckled in you stay buckled

**Comment by aceedex** [comment]
then that's a paddin'

**Comment by Dacendeth** [comment]
I think it healing just oxyloss while you're buckled would be reasonable.

**Comment by aceedex** [comment]
Alright, with the newest change the effect does not activate on any vehicle. This includes all wheelchairs, secways, the janicart and presumably the snowmobiles
As far as that suggestion dacendeth, its meant to help with anything that might cause considerable damage over time but isnt an active threat. This wont really do anything if youre eating laser to the face but it might let you hang on a little more if youre low on blood and suffering from oxyloss and toxins because of it.
also I dunno whats wrong with travis

**Comment by ShiftyRail** [comment]
It's a bit powerful
someone being in critical shouldn't require at least some machinery to revive

Travis is angry because you're not using 513 yet. Update your repo.

**Comment by Kammerjunk** [comment]
no, Travis is angry because it doesn't compile

File: code/modules/reagents/Chemistry-Reagents.dm [hunk 1]
```diff
@@ -1037,6 +1037,11 @@
 	else
 		if(M.losebreath >= 10)
 			M.losebreath = max(10, M.losebreath - 5)
+		if(M.InCritical() && !istype(M.locked_to,/obj/structure/bed/chair/vehicle)
```
Comment (Kammerjunk) [inline code/modules/reagents/Chemistry-Reagents.dm]: missing closing bracket

**Comment by boy2mantwicethefam** [comment]
Sure, why not

**Comment by ShiftyRail** [comment]
Kammer is correct apologies

**Comment by Kammerjunk** [comment]
also this will only work if they're buckled to a vehicle, which means they have to be in e.g. a wheelchair or a firebird or whatever, not a bed

**Comment by ArchSunder** [comment]
I actually do like this idea of making rollerbeds useful but I might agree with razz in making it oxy only

**Comment by aceedex** [comment]
You sure kammer? I used !istype and everything

I'd say reducing the healing is a suggestion Is be open to but the oxygen only thing is a little much in the opposite direction considering inaprov stops crit oxyloss by default 
I'd update my repo but I made my oldest PR on bleeding edge like an idiot

**Comment by Kammerjunk** [comment]
oh, sorry
what the if statement currently checks (assuming you add an additional closing bracket to actually close the condition) is
- is the mob in crit?
- is `locked_to` NOT of the vehicle type

what immediately comes to mind here is that `null` is not of type `/obj/structure/bed/chair/vehicle` and this might run if you aren't buckled to anything

you should probably make it compile and then check

**Comment by aceedex** [comment]
I did test this that exact way when it was just the syndie chair being excluded and it worked as it said on the tin, worked on all buckles, did not work when just in crit, did not work when it was just the chair. More than happy to test it again though

File: code/modules/reagents/Chemistry-Reagents.dm [hunk 2]
```diff
@@ -1037,6 +1037,11 @@
...
+				M.adjustOxyLoss(-2)
+				M.heal_organ_damage(2,2)
+				M.adjustToxLoss(-2)
```
Comment (Kammerjunk) [inline code/modules/reagents/Chemistry-Reagents.dm]: 2 tabs here instead of 1, causes compilation error as well

**Comment by Kammerjunk** [comment]
well here you go
image
just body: works
body buckled to chair: works
body buckled to bed: works
body in wheelchair: not works

it doesn't work in a wheelchair because wheelchair is a type of vehicle. it works when not buckled at all because `null` is not a type of vehicle

**Comment by aceedex** [comment]
🤨 huh. How would I exclude null as well?

**Comment by Kammerjunk** [comment]
well if you want it to work with *all* buckling except for with vehicles, you could do probably `M.locked_to && !istype(M.locked_to, /obj/structure/bed/chair/vehicle` and if you want it to work for beds, only beds, not parents of beds, and not subtypes of beds (like chairs or vehicles), you can just do `locked_to.type = /obj/structure/bed`

probably - I didn't test any of that

**Comment by aceedex** [comment]
Oh, I'm double retarded. I did originally have locked.to by itself but I think I lost it in one of the revisions. Thanks kammer

**Comment by Kurfursten** [comment]
Real talk: if you want inaprovaline to help stabilize people in unstable conditions and not just be a general cure-all for people under the crit threshold, why not just have it reduce or negate those destabilizing things directly?

For example, big killers on the way to medbay are suffocation due to blood loss or overtoxing due to liver damage. Directly interact with those things instead of broad healing everything.

I don't even know why a patient in transit would be taking constant fire or brute damage unless you're dragging them through a burning hallway or space which I don't think medicine should be designed for that situation.

**Comment by strawberrytan** [comment]
convert damage recovered this way into a good heaping of pain for when they're stable

**Comment by aceedex** [comment]
Well kurf, thats because there's many sources of damage over time in transport, and including them all would make it an actual multi purpose wonder drug. You mention IB and overdrinking which is all well and good, but there's many many things that can fill into that checkbox as well. Stuff like radiation (don't we have like 3 different radiations), poisons and all sort of floor pill goodies, busted lung and general organ damage, hypo and hyperthermia, being on fire, nonhumans without their snowflake gasses, breached and over pressurized rooms and so on so forth. The obvious question after that is, "Well why does it need all this why don't you just do one or two?" which is a very fair question. The answer being because then you'd be doing diagnosis on the spot instead of injecting it and taking the person to medbay or your safe spot of choice. 
Entertain that you find a passed out engineer with a ton of toxin damage in the hall. He could have drank his liver away, he could have just been irradiated to fuck by a shard dragger, coulda drank someone's blood, etc etc. Youre now playing a guessing game of "does inaprov even work and do enough for him to bother?"and I think that if the game was slower that'd be alright but its too fast to work as it is now. I reckon it wouldn't be something worth thinking about at all and you'd just take the person to medbay anyway. With the way its done now you just put the chem in because you'll know it'll do something and can leave the diagnosis work to actual doctors.

**Comment by Kurfursten** [comment]
I get what you're saying but what it amounts to is "you shouldn't have to diagnose this guy" which is less intelligent medbay play. It becomes a broad spectrum you throw into ANYBODY below crit, stick it in every pill why not [except Antitox of course, hue]? Inaprov is easy to mass produce, throw it in top of all the other things that heal all. And I mean really, are you hurting for inaprov so much that you feel uncertain about stabbing it into somebody? If time is critical, drag them straight there instead of fumbling to unfold your stetcher, injecting with delay, clicking to drag and leaving.

I think what I'm talking about here when I say "less intelligent medbay play" really is exemplified by the example you gave for burn damage. How about you **put their ass out with an extinguisher** instead of buffing inaprov to constantly heal burns so you can **drag your burning patient back to medbay and make a real doctor rub two braincells together**

**Comment by aceedex** [comment]
Kurf I specifically wrote
"because then you'd be doing diagnosis on the spot instead of injecting it and taking the person to medbay or your safe spot of choice."
No, it's not that I think diagnosis is dumb and shouldn't happen and medbay should just be bad number go down. It's that the doctors should be the ones doing that once the patient is delivered not whatever guy ends up being body recovery man. As to why not stick it to every pill? You could, just like what you can do with tricord and DD and stuff, it would still need to be used on someone buckled and in crit, but if you're in medbay you probably don't need to put someone in a bed and have them heal for a token amount when you can directly adress what's ailing them. 

The hypothetical of "should I inject" isn't about the supply of inaprov. It's about knowing if it did something or didn't, and how if you try to find the answer to that you're wasting so much time that you may as well have done nothing and body dragged in the first place. I agree though that the application is a little time consuming and if this change goes through I'll see how it's received and have a way to make the application more immediate  if needed, like IV roller beds that are foldable or a supply of inaprov autoinjectors, we'll have to see.

As far as that less intelligent medbay play bit I feel like you're assuming way too much. I didn't give just a single example for burn damage. Obviously if the person is on fire put them out but there's other sources where it isn't that simple. A plasmeme without their suit will just keep igniting, someone who was in a superheated area won't stop burning up for a bit when you take them out because of the celtian laws of hyperthermia, someone who ate a super spicy pepper won't stop burning for a long while because he'll keep metabolizing the spicy sauce, someone with a fever too, and I think acids or some other chems burn you up when ingested. many such cases.  That's why I went with this general heals damage apporach because covering every if would be silly and only covering some puts you into a guessing game situation where you can only lose time if you play it.

**Comment by Baronzegit** [comment]
Inaprovaline is already an extremely powerful chemical for some antag types as it is, giving it extra effects is not really helpful.

**Comment by aceedex** [comment]
I'm curious, what antag types and how? The goal here isnt to make it easier on antags and I think I got my bases covered with the vehicle exception

**Comment by Baronzegit** [comment]
> I'm curious, what antag types and how? The goal here isnt to make it easier on antags and I think I got my bases covered with the vehicle exception

For example wizard, inaprovaline being readily available around the station allows you to go into crit but come back with some basic meds. Same goes for any time people have to go into combat, inaprovaline is a painkiller and it stops oxygen loss(for cases where you have blood loss, have enough meds in you but go into crit etc).

Generally speaking it is a very useful med, but buffing it would make it far too powerful.

**Comment by 9percents** [comment]
just buffing inaprovaline seems strange, it already has a good stabilizing power with removing breathloss and slowing down bleeding(inaprovaline autoinjectors when)

I would be all for this PR if it made a new medicine with these effects
more things to make for chemistry that has usage is good

**Comment by ShiftyRail** [comment]
I think this isn't going to pass in its current form.

**Pull request closed without merging**